media/starboard/sbplayer_bridge.cc

```diff
@@ -230,38 +203,50 @@ SbPlayerBridge::SbPlayerBridge(
     SbPlayerSetBoundsHelper* set_bounds_helper,
     bool allow_resume_after_suspend,
     SbPlayerOutputMode default_output_mode,
+#if COBALT_MEDIA_ENABLE_DECODE_TARGET_PROVIDER
     DecodeTargetProvider* const decode_target_provider,
+#endif  // COBALT_MEDIA_ENABLE_DECODE_TARGET_PROVIDER
     const std::string& max_video_capabilities,
-    int max_video_input_size,
-    std::string pipeline_identifier)
+    int max_video_input_size
+#if COBALT_MEDIA_ENABLE_CVAL
+    ,
+    std::string pipeline_identifier
+#endif  // COBALT_MEDIA_ENABLE_CVAL
+    )
     : sbplayer_interface_(interface),
       task_runner_(task_runner),
       get_decode_target_graphics_context_provider_func_(
           get_decode_target_graphics_context_provider_func),
-      callback_helper_(
-          new CallbackHelper(ALLOW_THIS_IN_INITIALIZER_LIST(this))),
+      callback_helper_(new CallbackHelper(this)),
       window_(window),
       drm_system_(drm_system),
       host_(host),
       set_bounds_helper_(set_bounds_helper),
       allow_resume_after_suspend_(allow_resume_after_suspend),
       audio_config_(audio_config),
       video_config_(video_config),
+#if COBALT_MEDIA_ENABLE_DECODE_TARGET_PROVIDER
       decode_target_provider_(decode_target_provider),
+#endif  // COBALT_MEDIA_ENABLE_DECODE_TARGET_PROVIDER
       max_video_capabilities_(max_video_capabilities),
-      max_video_input_size_(max_video_input_size),
-      cval_stats_(&interface->cval_stats_),
-      pipeline_identifier_(pipeline_identifier)
+      max_video_input_size_(max_video_input_size)
+#if COBALT_MEDIA_ENABLE_CVAL
+          cval_stats_(&interface->cval_stats_),
+      pipeline_identifier_(pipeline_identifier),
+#endif  // COBALT_MEDIA_ENABLE_CVAL
 #if SB_HAS(PLAYER_WITH_URL)
-      ,
```

Comment:
The comma should be kept.

Reply:
The code probably won't build whether the comma is there.  I'll keep it removed (so the code is cleaner now) and address it when working on url player later.